src/java.base/share/classes/java/lang/classfile/Opcode.java

```diff
@@ -24,684 +24,1634 @@
+    /**
+     * Push {@code null}.
+     *
+     * @jvms 6.5.aconst_null <em>aconst_null</em>
+     * @see ConstantInstruction.IntrinsicConstantInstruction
+     */
     ACONST_NULL(RawBytecodeHelper.ACONST_NULL, 1, Kind.CONSTANT),
 
-    /** Push int constant -1 */
+    /**
+     * Push {@link TypeKind#INT int} constant {@code -1}.
+     *
+     * @jvms 6.5.iconst_i <em>iconst_&lt;i&gt;</em>
+     * @see ConstantInstruction.IntrinsicConstantInstruction
+     */
     ICONST_M1(RawBytecodeHelper.ICONST_M1, 1, Kind.CONSTANT),
 
-    /** Push int constant 0 */
+    /**
+     * Push {@link TypeKind#INT int} constant {@code 0}.
+     *
+     * @jvms 6.5.iconst_i <em>iconst_&lt;i&gt;</em>
+     * @see ConstantInstruction.IntrinsicConstantInstruction
+     */
     ICONST_0(RawBytecodeHelper.ICONST_0, 1, Kind.CONSTANT),
 
-    /** Push int constant 1 */
+    /**
+     * Push {@link TypeKind#INT int} constant {@code 1}.
+     *
+     * @jvms 6.5.iconst_i <em>iconst_&lt;i&gt;</em>
+     * @see ConstantInstruction.IntrinsicConstantInstruction
+     */
     ICONST_1(RawBytecodeHelper.ICONST_1, 1, Kind.CONSTANT),
 
-    /** Push int constant 2 */
+    /**
+     * Push {@link TypeKind#INT int} constant {@code 2}.
+     *
+     * @jvms 6.5.iconst_i <em>iconst_&lt;i&gt;</em>
+     * @see ConstantInstruction.IntrinsicConstantInstruction
+     */
     ICONST_2(RawBytecodeHelper.ICONST_2, 1, Kind.CONSTANT),
 
-    /** Push int constant 3 */
+    /**
+     * Push {@link TypeKind#INT int} constant {@code 3}.
+     *
+     * @jvms 6.5.iconst_i <em>iconst_&lt;i&gt;</em>
+     * @see ConstantInstruction.IntrinsicConstantInstruction
+     */
     ICONST_3(RawBytecodeHelper.ICONST_3, 1, Kind.CONSTANT),
 
-    /** Push int constant 4 */
+    /**
+     * Push {@link TypeKind#INT int} constant {@code 4}.
+     *
+     * @jvms 6.5.iconst_i <em>iconst_&lt;i&gt;</em>
+     * @see ConstantInstruction.IntrinsicConstantInstruction
+     */
     ICONST_4(RawBytecodeHelper.ICONST_4, 1, Kind.CONSTANT),
 
-    /** Push int constant 5 */
+    /**
+     * Push {@link TypeKind#INT int} constant {@code 5}.
+     *
+     * @jvms 6.5.iconst_i <em>iconst_&lt;i&gt;</em>
+     * @see ConstantInstruction.IntrinsicConstantInstruction
+     */
     ICONST_5(RawBytecodeHelper.ICONST_5, 1, Kind.CONSTANT),
 
-    /** Push long constant 0 */
+    /**
+     * Push {@link TypeKind#LONG long} constant {@code 0}.
+     *
+     * @jvms 6.5.lconst_l <em>lconst_&lt;l&gt;</em>
+     * @see ConstantInstruction.IntrinsicConstantInstruction
+     */
     LCONST_0(RawBytecodeHelper.LCONST_0, 1, Kind.CONSTANT),
 
-    /** Push long constant  1 */
+    /**
+     * Push {@link TypeKind#LONG long} constant {@code 1}.
+     *
+     * @jvms 6.5.lconst_l <em>lconst_&lt;l&gt;</em>
+     * @see ConstantInstruction.IntrinsicConstantInstruction
+     */
     LCONST_1(RawBytecodeHelper.LCONST_1, 1, Kind.CONSTANT),
 
-    /** Push float constant 0 */
+    /**
+     * Push {@link TypeKind#FLOAT float} constant {@code 0}.
```

Comment:
For the float and double constants, it would be a small improvement to write them out as floating-point literals, "0.0f" for a float zero, etc.